src/alert/alert.ts

```diff
@@ -38,12 +43,22 @@ export class NgbAlert {
   /**
    * An event emitted when the close button is clicked. This event has no payload. Only relevant for dismissible alerts.
    */
-  @Output() close = new EventEmitter();
+  @Output() close = new EventEmitter<void>();
 
-  constructor(config: NgbAlertConfig) {
+  constructor(config: NgbAlertConfig, private _renderer: Renderer2, private _element: ElementRef) {
     this.dismissible = config.dismissible;
     this.type = config.type;
   }
 
   closeHandler() { this.close.emit(null); }
+
+  ngOnChanges(changes: SimpleChanges) {
+    const {type} = changes;
```

Comment:
While it works I find the `type` name confusing / not very descriptive  here. A much better name would be `typeChange`. I would prefer to be explicit here.

Reply:
Will change